File: Asistencia/Asistencia.R
## @knitr inic
#Jose Miguel Saavedra Aguilar
library(ordinal)
set.seed(157017)

## @knitr t1inicializacion
# Creamos un vector de 100 puntos en el soporte.
# En nuestro caso, tomamos de 0 a 10.
x <- seq(0, 10, length = 100)

## @knitr t1ej1
# y es la función de densidad elegida evaluada en los puntos de x
# Para obtener la función de densidad, se toma "d"+nombre de la distribución,
# por ejemplo dnorm o dexp
y1 <- dexp(x, rate = 5)
y2 <- dexp(x, rate = 0.5)
y3 <- dexp(x, rate = 2)


## @knitr t1ej2
# f es la función de distribución asociada evaluada en los puntos de x
# Para obtener la función de densidad, se toma "p"+nombre de la distribución,
# por ejemplo pnorm o pexp.
f1 <- pexp(x, rate = 5)
f2 <- pexp(x, rate = 0.5)
f3 <- pexp(x, rate = 2)


## @knitr t1ej3
# Tomamos n=100 y simulamos n datos pseudoaleatorios de la distribución elegida
# Para obtener los datos aleatorios, se toma "r"+nombre de la distribución,
# por ejemplo rnorm o rexp
n <- 100
z <- rexp(n, rate = 0.5)

## @knitr t1ej4
#Ahora vamos a ordenar los datos como indica la tarea.
z2 <- sort(z)
# Inicializamos k =i/n+1
k <- numeric()

for (i in 1:n) {
  k[i] <- i / (n + 1)
}

## @knitr t1gr1
plot(x, y1, type = "l", col = "red", xlab = "x", ylab = "f(x)")
lines(x, y2, col = "blue")
lines(x, y3, col = "green")

## @knitr t1gr2
plot(x, f1, type = "l", col = "red", xlab = "x", ylab = "f(x)")
lines(x, f2, col = "blue")
lines(x, f3, col = "green")

## @knitr t1gr3
hist(z, breaks = "FD", freq = FALSE)
lines(x, y2)

## @knitr t1gr4
#Graficamos los puntos y comparamos con la función de densidad teórica
plot(z2, k, col = "red", type = "l")
lines(x, f2, col = "blue")

## @knitr t2ej1
#Ejemplo 2
set.seed(157017)
prob <- c(.1, 0.3, 0.5, 0.7, 1)
frec <- findInterval(runif(1000), prob)
table(frec) / 1000
set.seed(157017)
frecSample <- sample(x = 0:4, size = 1000, replace = TRUE,
                     prob = c(0.1, 0.2, 0.2, 0.2, 0.3))
table(frecSample) / 1000

## @knitr t2ej2
#Ejemplo 3
p <- function(n) {
  values <- sample(1:100, n, replace = TRUE)
  prob <- (2 * values) / (100 * 101)
  return(prob)
}
head(p(10000), n = 30)

## @knitr t2ej3
#Ejemplo 4
mayor <- 0
for (i in 1:10000){
  reclamaciones <- sum(rbinom(1000, 1, 0.09245))
  montos <- sum(rgamma(reclamaciones, 7000, 1))
  if (montos > 500000) {
    mayor <- mayor + 1
  }
}
mayor / 10000


## @knitr t3estimadores
est1 <- function(x) {
  mean(x)
}

est2 <- function(x) {
  n <- length(x)
  sum((x - mean(x))^2) / n
}

est3 <- function(x) {
  n <- length(x)
  -0.5 + sqrt(.25 + sum(x^2) / n)
}

## @knitr t3ej2
lambda <- 3
n1 <- 10
n2 <- 20
n3 <- 40
n4 <- 80
n5 <- 200

x1 <- rpois(n1, lambda)
x2 <- rpois(n2, lambda)
x3 <- rpois(n3, lambda)
x4 <- rpois(n4, lambda)
x5 <- rpois(n5, lambda)

e11 <- est1(x1)
e12 <- est2(x1)
e13 <- est3(x1)
e21 <- est1(x2)
e22 <- est2(x2)
e23 <- est3(x2)
e31 <- est1(x3)
e32 <- est2(x3)
e33 <- est3(x3)
e41 <- est1(x4)
e42 <- est2(x4)
e43 <- est3(x4)
e51 <- est1(x5)
e52 <- est2(x5)
e53 <- est3(x5)

print("  n |  est1,  est2,  est3")
print(sprintf("%3i | %1.3f, %1.3f, %1.3f", n1, e11, e12, e13))
print(sprintf("%3i | %1.3f, %1.3f, %1.3f", n2, e21, e22, e23))
print(sprintf("%3i | %1.3f, %1.3f, %1.3f", n3, e31, e32, e33))
print(sprintf("%3i | %1.3f, %1.3f, %1.3f", n4, e41, e42, e43))
print(sprintf("%3i | %1.3f, %1.3f, %1.3f", n5, e51, e52, e53))


## @knitr t3ej3dfn
n1 <- 15
x1 <- rnorm(n1, mean = 60, sd = 5)
k1 <- 1:n1 / (n1)


## @knitr t3ej3a

estMu <- function(x) {
  mean(x)
}

estSigma <- function(x) {
  n <- length(x)
  sqrt(sum((x - mean(x))^2) / n)
}

mu1 <- estMu(x1)
sigma1 <- estSigma(x1)
k1 <- 1 : n1 / (n1)

## @knitr t3ej3b
x <- seq(40, 80, length = 500)
plot(x, pnorm(x, mean = 60, sd = 5), type = "l", col = "red",
     xlab = "x", ylab = "F(x)")
lines(x, pnorm(x, mean = mu1, sd = sigma1), lty = 2, col = "blue")
lines(c(0, sort(x1), 100), c(0,k1,1), col = "green", type = "s")
legend("bottomright", legend = c("Densidad teórica", "Densidad estimada",
                                 "Densidad Empírica"),
       col = c("red", "blue", "green"), lty = c(1, 2, 1))

## @knitr t3ej3dfn2
n2 <- 30
x2 <- rnorm(n2, mean = 60, sd = 5)
n3 <- 100
x3 <- rnorm(n3, mean = 60, sd = 5)
k2 <- 1:n2 / (n2)
k3 <- 1:n3 / (n3)

## @knitr t3ej3c
mu2 <- estMu(x2)
sigma2 <- estSigma(x2)
mu3 <- estMu(x3)
sigma3 <- estSigma(x3)

## @knitr t3ej3c1
plot(x, pnorm(x, mean = 60, sd = 5), type = "l", col = "red",
     xlab = "x", ylab = "F(x)")
lines(x, pnorm(x, mean = mu2, sd = sigma2), lty = 2, col = "blue")
lines(c(0, sort(x2), 100), c(0, k2, 1), col = "green", type = "s")
legend("bottomright", legend = c("Densidad teórica", "Densidad estimada",
                                 "Densidad Empírica"),
       col = c("red", "blue", "green"), lty = c(1, 2, 1))

## @knitr t3ej3c2
plot(x, pnorm(x, mean = 60, sd = 5), type = "l", col = "red", xlab = "x",
     ylab = "F(x)")
lines(x, pnorm(x, mean = mu3, sd = sigma3), lty = 2, col = "blue")
lines(c(0, sort(x3), 100), c(0, k3, 1), col = "green", type = "s")
legend("bottomright",
       legend = c("Densidad teórica", "Densidad estimada", "Densidad Empírica"),
       col = c("red", "blue", "green"), lty = c(1, 2, 1))


## @knitr t4setBootstrap
n <- 300
lambda <- 5
x <- rexp(300, rate = lambda)


## @knitr t4bootstrap
r <- mean(x)
k <- 100
m <- 30
q1 <- 0.025
q2 <- 0.975

q1Est <- numeric(k)
q2Est <- numeric(k)

for (i in 1:k){
  z <- sample(x,size = m, replace = FALSE)
  q1Est[i] <- quantile(z,q1)
  q2Est[i] <- quantile(z,q2)
}

## @knitr t4comparativo

Q1 <- mean(q1Est)
Q2 <- mean(q2Est)

S1 <- sd(q1Est)
S2 <- sd(q2Est)

print(sprintf("La media del cuantil %1.3f es %2.3f", q1, Q1))
print(sprintf("La media del cuantil %1.3f es %2.3f", q2, Q2))
print(sprintf("La desviación estándar del cuantil %1.3f es %2.3f", q1, S1))
print(sprintf("La desviación estándar del cuantil %1.3f es %2.3f", q2, S2))

## @knitr t5setBootstrap
n <- 300
lambda <- 5
x <- rexp(300, rate = lambda)


## @knitr t5bootstrap
r <- mean(x)
k <- 100
m <- 30
q1 <- 0.025
q2 <- 0.975

q1Est <- numeric(k)
q2Est <- numeric(k)

for (i in 1:k){
  z <- rexp(m, rate = 1 / r)
  rBoot <- mean(z)
  q1Est[i] <- qexp(q1, rate = 1 / rBoot)
  q2Est[i] <- qexp(q2, rate = 1 / rBoot)
}

## @knitr t5comparativo

Q1 <- mean(q1Est)
Q2 <- mean(q2Est)

S1 <- sd(q1Est)
S2 <- sd(q2Est)

print(sprintf("La media del cuantil %1.3f es %2.3f", q1, Q1))
print(sprintf("La media del cuantil %1.3f es %2.3f", q2, Q2))
print(sprintf("La desviación estándar del cuantil %1.3f es %2.3f", q1, S1))
print(sprintf("La desviación estándar del cuantil %1.3f es %2.3f", q2, S2))

## @knitr t6setBootstrap
n <- 300
lambda <- 5
x <- rexp(300, rate = lambda)


## @knitr t6bootstrap
r <- mean(x)
k <- 100
m <- 30
q1 <- 0.025
q2 <- 0.975

q1Est <- numeric(k)
q2Est <- numeric(k)

for (i in 1:k){
  z <- rexp(m, rate = 1 / r)
  rBoot <- mean(z)
  q1Est[i] <- qexp(q1, rate = 1 / rBoot)
  q2Est[i] <- qexp(q2, rate = 1 / rBoot)
}

## @knitr t6normal

Q1 <- mean(q1Est)
Q2 <- mean(q2Est)

S1 <- sd(q1Est)  # R uses ddof=1 by default for unbiased standard deviation
S2 <- sd(q2Est)

T1L <- Q1 - S1 * qnorm(0.975)
T1R <- Q1 + S1 * qnorm(0.975)

T2L <- Q2 - S2 * qnorm(0.975)
T2R <- Q2 + S2 * qnorm(0.975)

q1_real <- qexp(q1, rate = lambda)
q2_real <- qexp(q2, rate = lambda)

print("Por el método normal:")
print(sprintf("El intervalo de confianza para el cuantil %.3f es (%.5f, %.5f)", q1, T1L, T1R))
if (T1L < q1_real && q1_real < T1R) {
  print(sprintf("El cuantil %.3f está en el intervalo de confianza", q1))
} else {
  print(sprintf("El cuantil %.3f no está en el intervalo de confianza", q1))
}

print(sprintf("El intervalo de confianza para el cuantil %.3f es (%.5f, %.5f)", q2, T2L, T2R))
if (T2L < q2_real && q2_real < T2R) {
  print(sprintf("El cuantil %.3f está en el intervalo de confianza", q2))
} else {
  print(sprintf("El cuantil %.3f no está en el intervalo de confianza", q2))
}

## @knitr t6pivotal

q1_N <- qexp(q1, rate = 1 / r)
q2_N <- qexp(q2, rate = 1 / r)

q1_R <- quantile(q1Est, 0.025)
q1_L <- quantile(q1Est, 0.975)
q2_R <- quantile(q2Est, 0.025)
q2_L <- quantile(q2Est, 0.975)

P1L <- 2 * q1_N - q1_L
P1R <- 2 * q1_N - q1_R
P2L <- 2 * q2_N - q2_L
P2R <- 2 * q2_N - q2_R

print("Por el método pivotal:")
print(sprintf("El intervalo de confianza para el cuantil %.3f es (%.5f, %.5f)", q1, P1L, P1R))
if (P1L < q1_real && q1_real < P1R) {
  print(sprintf("El cuantil %.3f está en el intervalo de confianza", q1))
} else {
  print(sprintf("El cuantil %.3f no está en el intervalo de confianza", q1))
}

print(sprintf("El intervalo de confianza para el cuantil %.3f es (%.5f, %.5f)", q2, P2L, P2R))
if (P2L < q2_real && q2_real < P2R) {
  print(sprintf("El cuantil %.3f está en el intervalo de confianza", q2))
} else {
  print(sprintf("El cuantil %.3f no está en el intervalo de confianza", q2))
}

File: Asistencia/Asistencia.py
# Jose Miguel Saavedra Aguilar
import random
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import expon, norm

random.seed(157017)

# Creamos un vector de 100 puntos en el soporte.
# En nuestro caso, tomamos de 0 a 10.
x = np.linspace(0, 10, num=100)

# y es la función de densidad elegida evaluada en los puntos de x
# Para obtener la función de densidad, se toma "pdf" + nombre de la distribución,
# por ejemplo expon.pdf
y1 = expon.pdf(x, scale=1/5)
y2 = expon.pdf(x, scale=1/0.5)
y3 = expon.pdf(x, scale=1/2)

# f es la función de distribución asociada evaluada en los puntos de x
# Para obtener la función de distribución, se toma "cdf" + nombre de la distribución,
# por ejemplo expon.cdf.
f1 = expon.cdf(x, scale=1/5)
f2 = expon.cdf(x, scale=1/0.5)
f3 = expon.cdf(x, scale=1/2)

# Tomamos n=100 y simulamos n datos pseudoaleatorios de la distribución elegida
# Para obtener los datos aleatorios, se toma "rvs" + nombre de la distribución,
# por ejemplo expon.rvs
n = 100
z = expon.rvs(scale=1/0.5, size=n)

# Ordenamos los datos como indica la tarea
z2 = np.sort(z)

# Inicializamos k = i/n+1
k = np.zeros(n)
for i in range(n):
    k[i] = (i+1) / (n+1)

# Gráficas
# Gráfico 1
plt.figure()
plt.plot(x, y1, 'red', label='rate = 5')
plt.plot(x, y2, 'blue', label='rate = 0.5')
plt.plot(x, y3, 'green', label='rate = 2')
plt.xlabel('x')
plt.ylabel('f(x)')
plt.legend()
plt.show()

# Gráfico 2
plt.figure()
plt.plot(x, f1, 'red', label='rate = 5')
plt.plot(x, f2, 'blue', label='rate = 0.5')
plt.plot(x, f3, 'green', label='rate = 2')
plt.xlabel('x')
plt.ylabel('F(x)')
plt.legend()
plt.show()

# Gráfico 3
plt.figure()
plt.hist(z, bins='auto', density=True)
plt.plot(x, y2)
plt.xlabel('x')
plt.ylabel('Frequency')
plt.title('Histogram of z')
plt.show()

# Gráfico 4
plt.figure()
plt.plot(z2, k, 'red')
plt.plot(x, f2, 'blue')
plt.xlabel('z')
plt.ylabel('k')
plt.title('Empirical Quantile Function')
plt.tight_layout()
plt.show()


# Ejemplo 2

prob = [0.1, 0.3, 0.5, 0.7, 1]
frec = np.histogram(np.random.uniform(size=1000), bins=np.cumsum(prob))[0] / 1000

np.random.seed(157017)
frecSample = np.histogram(np.random.choice(np.arange(5), size=1000, replace=True, p=[0.1, 0.2, 0.2, 0.2, 0.3]))[0] / 1000

print(f"Ejemplo 2:\n{frec}")
print(f"Ejemplo 2 (using sample):\n{frecSample}")

# Ejemplo 3
def p(n):
    values = np.random.choice(np.arange(1, 101), size=n, replace=True)
    prob = (2 * values) / (100 * 101)
    return prob

print("\nEjemplo 3:")
print(p(10000)[:30])

# Ejemplo 4
mayor = sum(np.sum(np.random.binomial(1, 0.09245, 1000)) for _ in range(10000))
montos = sum(np.sum(np.random.gamma(7000, 1, reclamaciones)) for reclamaciones in np.random.binomial(1000, 0.09245, 10000))
result = mayor / 10000
print(f"\nEjemplo 4:\n{result}")

# Tarea 3
# Ejercicio 1. Estimadores
def est1(x):
    return np.mean(x)

def est2(x):
    n = len(x)
    return np.sum((x - np.mean(x))**2) / n

def est3(x):
    n = len(x)
    return -0.5 + np.sqrt(0.25 + np.sum(x**2) / n)

# Ejercicio 1. Simulación
lambda_val = 3
n_values = [10, 20, 40, 80, 200]

for n in n_values:
    x = np.random.poisson(lambda_val, n)
    e1 = est1(x)
    e2 = est2(x)
    e3 = est3(x)
    print(f"n = {n:3} | est1: {e1:.3f}, est2: {e2:.3f}, est3: {e3:.3f}")

# Ejercicio 2. Definiciones
n1 = 15
x1 = np.random.normal(loc=60, scale=5, size=n1)
k1 = np.arange(1, n1 + 1) / n1

# Ejercicio 2. Estimadores
mu1 = est1(x1)
sigma1 = est2(x1)

# Ejercicio 2. Gráficas
x = np.linspace(40, 80, 500)
plt.figure()
plt.plot(x, norm.cdf(x, loc=60, scale=5), label="Densidad teórica", color="red")
plt.plot(x, norm.cdf(x, loc=mu1, scale=sigma1), label="Densidad estimada", linestyle="--", color="blue")
plt.step(np.concatenate(([0], np.sort(x1), [100])), np.concatenate(([0], k1, [1])), label="Densidad Empírica", color="green", linestyle="-.")
plt.xlabel("x")
plt.ylabel("F(x)")
plt.legend()
plt.show()

# Ejercicio 2b. Definiciones
n2 = 30
x2 = np.random.normal(loc=60, scale=5, size=n2)
n3 = 100
x3 = np.random.normal(loc=60, scale=5, size=n3)
k2 = np.arange(1, n2 + 1) / n2
k3 = np.arange(1, n3 + 1) / n3

# Ejercicio 2b. Estimadores
mu2, sigma2 = est1(x2), est2(x2)
mu3, sigma3 = est1(x3), est2(x3)

# Ejercicio 2c. Gráfica 1
plt.figure()
plt.plot(x, norm.cdf(x, loc=60, scale=5), label="Densidad teórica", color="red")
plt.plot(x, norm.cdf(x, loc=mu2, scale=sigma2), label="Densidad estimada", linestyle="--", color="blue")
plt.step(np.concatenate(([0], np.sort(x2), [100])), np.concatenate(([0], k2, [1])), label="Densidad Empírica", color="green", linestyle="-.")
plt.xlabel("x")
plt.ylabel("F(x)")
plt.legend()
plt.show()

# Ejercicio 2c. Gráfica 2
plt.figure()
plt.plot(x, norm.cdf(x, loc=60, scale=5), label="Densidad teórica", color="red")
plt.plot(x, norm.cdf(x, loc=mu3, scale=sigma3), label="Densidad estimada", linestyle="--", color="blue")
plt.step(np.concatenate(([0], np.sort(x3), [100])), np.concatenate(([0], k3, [1])), label="Densidad Empírica", color="green", linestyle="-.")
plt.xlabel("x")
plt.ylabel("F(x)")
plt.legend()
plt.show()


# Bootstrap no paramétrico. Parámetros de la distribución real
n = 300
lambda_val = 5
x = np.random.exponential(scale=1/lambda_val, size=n)

# Bootstrap no paramétrico. Parámetros de bootstrap
r = np.mean(x)
k = 100
m = 30
q1 = 0.025
q2 = 0.975

q1_est = np.zeros(k)
q2_est = np.zeros(k)

# Bootstrap no paramétrico. Bootstrap

for i in range(k):
    z = np.random.choice(x, size=m, replace=False)
    q1_est[i] = np.percentile(z, q=q1)
    q2_est[i] = np.percentile(z, q=q2)

# Bootstrap no paramétrico. Resultados
Q1 = np.mean(q1_est)
Q2 = np.mean(q2_est)

S1 = np.std(q1_est, ddof=1)  # ddof=1 for unbiased standard deviation
S2 = np.std(q2_est, ddof=1)

print(f"La media del cuantil {q1:.3f} es {Q1:.3f}")
print(f"La media del cuantil {q2:.3f} es {Q2:.3f}")
print(f"La desviación estándar del cuantil {q1:.3f} es {S1:.3f}")
print(f"La desviación estándar del cuantil {q2:.3f} es {S2:.3f}")

# Bootstrap paramétrico. Parámetros de la distribución real

n = 300
lambda_val = 5
x = np.random.exponential(scale=1/lambda_val, size=n)

# Bootstrap paramétrico. Parámetros de bootstrap
r = np.mean(x)
k = 100
m = 30
q1 = 0.025
q2 = 0.975

q1_est = np.zeros(k)
q2_est = np.zeros(k)

# Bootstrap paramétrico. Bootstrap
for i in range(k):
    z = np.random.exponential(scale=r, size=m)
    r_boot = np.mean(z)
    q1_est[i] = expon.ppf(q1, scale=r_boot)
    q2_est[i] = expon.ppf(q2, scale=r_boot)

# Bootstrap paramétrico. Resultados

Q1 = np.mean(q1_est)
Q2 = np.mean(q2_est)

S1 = np.std(q1_est, ddof=1)  # ddof=1 for unbiased standard deviation
S2 = np.std(q2_est, ddof=1)

print(f"La media del cuantil {q1:.3f} es {Q1:.3f}")
print(f"La media del cuantil {q2:.3f} es {Q2:.3f}")
print(f"La desviación estándar del cuantil {q1:.3f} es {S1:.3f}")
print(f"La desviación estándar del cuantil {q2:.3f} es {S2:.3f}")


# Bootstrap paramétrico. Parámetros de la distribución real
n = 300
lambda_val = 5
x = np.random.exponential(scale=1/lambda_val, size=n)

# Bootstrap paramétrico. Parámetros de bootstrap
r = np.mean(x)
k = 1000
m = 100
q1 = 0.025
q2 = 0.975

q1_est = np.zeros(k)
q2_est = np.zeros(k)

# Bootstrap paramétrico. Bootstrap

for i in range(1, k+1):
    z = np.random.exponential(scale=r, size=m)
    r_boot = np.mean(z)
    q1_est[i-1] = expon.ppf(q1, scale=r_boot)
    q2_est[i-1] = expon.ppf(q2, scale=r_boot)

# Bootstrap paramétrico. Resultados incluyendo intervalos de confianza
Q1 = np.mean(q1_est)
Q2 = np.mean(q2_est)

S1 = np.std(q1_est, ddof=1)  # ddof=1 for unbiased standard deviation
S2 = np.std(q2_est, ddof=1)

T1L = Q1 - S1 * norm.ppf(0.975)
T1R = Q1 + S1 * norm.ppf(0.975)

T2L = Q2 - S2 * norm.ppf(0.975)
T2R = Q2 + S2 * norm.ppf(0.975)

q1_real = expon.ppf(q1, scale=1/lambda_val)
q2_real = expon.ppf(q2, scale=1/lambda_val)

print("Por el método normal:")
print(f"El intervalo de confianza para el cuantil {q1:.3f} es ({T1L:.5f}, {T1R:.5f})")
if T1L < q1_real < T1R:
    print(f"El cuantil {q1:.3f} está en el intervalo de confianza")
else:
    print(f"El cuantil {q1:.3f} no está en el intervalo de confianza")

print(f"El intervalo de confianza para el cuantil {q2:.3f} es ({T2L:.5f}, {T2R:.5f})")
if T2L < q2_real < T2R:
    print(f"El cuantil {q2:.3f} está en el intervalo de confianza")
else:
    print(f"El cuantil {q2:.3f} no está en el intervalo de confianza")

q1_N = expon.ppf(q1, scale=r)
q2_N = expon.ppf(q2, scale=r)

q1_R = np.percentile(q1_est, q=2.5)
q1_L = np.percentile(q1_est, q=97.5)
q2_R = np.percentile(q2_est, q=2.5)
q2_L = np.percentile(q2_est, q=97.5)

P1L = 2 * q1_N - q1_L
P1R = 2 * q1_N - q1_R
P2L = 2 * q2_N - q2_L
P2R = 2 * q2_N - q2_R

print("Por el método pivotal:")
print(f"El intervalo de confianza para el cuantil {q1:.3f} es ({P1L:.5f}, {P1R:.5f})")
if P1L < q1_real < P1R:
    print(f"El cuantil {q1:.3f} está en el intervalo de confianza")
else:
    print(f"El cuantil {q1:.3f} no está en el intervalo de confianza")

print(f"El intervalo de confianza para el cuantil {q2:.3f} es ({P2L:.5f}, {P2R:.5f})")
if P2L < q2_real < P2R:
    print(f"El cuantil {q2:.3f} está en el intervalo de confianza")
else:
    print(f"El cuantil {q2:.3f} no está en el intervalo de confianza")